### DISCUSSION TITLE
Compatibility With Overleaf Community 5.0.4

### DISCUSSION THREAD

**Discussion by Mingbo-Lee** [overleaf-workshop/Overleaf-Workshop#150] [General]
Thank you so much for your work that made it possible for us to use Overleaf on VS Code, it's awesome!

I've noticed that in your current plugin compatibility experiment, the version is still stuck at 4.2, while the latest version is 5.0.4. I am very happy to inform you that I have deployed the latest version of overleaf, and using your plugin to connect, is actually feasible, I will give my steps below for your reference.

- I'll first refer to [here](https://github.com/overleaf/toolkit/blob/master/doc/quick-start-guide.md), execute the command:
```bash
git clone https://github.com/overleaf/toolkit.git ./overleaf-toolkit
```
```bash
cd ./overleaf-toolkit
```
```bash
bin/init
```
- Then I modified the `overleaf-toolkit/lib/docker-compose.base.yml` file according to my own needs
```
image: "${IMAGE}"  -->   image: tuetenk0pp/sharelatex-full
```
I don't think the additional operation of [tuetenk0pp](https://github.com/tuetenk0pp/sharelatex-full/tree/master) will affect the version compatibility of your plugin.
- Finally, I modified the file `overleaf-toolkit/config/overleaf.rc`
```
OVERLEAF_LISTEN_IP=127.0.0.1 --> OVERLEAF_LISTEN_IP=0.0.0.0  
```
Add a comment to TLS proxy configuration, such as 
```
# TLS proxy configuration (optional)
# See documentation in doc/tls-proxy.md
NGINX_ENABLED=false
# NGINX_CONFIG_PATH=config/nginx/nginx.conf
# NGINX_HTTP_PORT=80
# Replace these IP addresses with the external IP address of your host
# NGINX_HTTP_LISTEN_IP=127.0.1.1
# NGINX_TLS_LISTEN_IP=127.0.1.1
# TLS_PRIVATE_KEY_PATH=config/nginx/certs/overleaf_key.pem
# TLS_CERTIFICATE_PATH=config/nginx/certs/overleaf_certificate.pem
# TLS_PORT=443
```
- Finally I start the cluster
```
bin/up
```
Some pictures for your reference：
![image](https://github.com/iamhyc/Overleaf-Workshop/assets/51162830/8f5b3224-2cc1-452a-9b1a-365db36058b9)
![image](https://github.com/iamhyc/Overleaf-Workshop/assets/51162830/a14fcb21-bfa9-4159-a49b-a66b9663d8fe)

I can submit a PR if you need it, and I look forward to hearing from you.

**Comment by iamhyc** [comment]
Thanks for your feedback! You can definitely create a PR to update the compatibility section.

**Comment by Mingbo-Lee** [comment]
Please review PR https://github.com/iamhyc/Overleaf-Workshop/pull/151
